Updates

jeikabu opened the pull request:
- ESP-IDF to v4.0
- LLVM to 9.0.1
- Rust to 1.42

jeikabu committed:
Updates

- ESP-IDF to v3.3.2
- LLVM to 9.0.1
- Rust to 1.42

jeikabu committed:
ESP-IDF v4.0

jeikabu committed:
Changes for ESP-IDF v4.0

- Before source'ing $IDF_PATH/export.sh, check if IDF_TOOLS_EXPORT_CMD is unset.  This means it hasn't been source'd yet.
- image-project assumes a particular target, config, and binary name.  If image.sh exists, assume that is a more appropriate replacement and run it instead.
- quick-build allows for quicker build iteration with less "safety"
- --release is no longer required, forward command-line flags like in PR #12 since it works just as well for --features as --release

jeikabu committed:
Update llvm, minor re-org of args, use xbuild 0.5.29

- Need to avoid LTO changes that result in gcc linking with compiler_builtins.rlib `error adding symbols: file format not recognized`

bvanderveen commented:
Master branch currently isn't working for me. This branch is. FWIW. I don't remember the exact error message but at some point in the past two weeks, I cleaned and rebuilt my projects and `build-project` failed before getting to my code. I ended up at this issue: https://github.com/rust-osdev/cargo-xbuild/pull/75/

jeikabu commented:
Yeah it's a bug with xbuild. You'll have to rebuild the docker image because the one on quay has an older xbuild.
And to clarify the specific version of xbuild used here, the latest versions use llvm features not yet ported.
Further, if it's helpful, as mentioned in https://github.com/rust-osdev/cargo-xbuild/issues/72 this [very similar project](https://github.com/rbtying/rust-xtensa-docker) may permit you to use the latest xbuild.

jeikabu committed:
ESP-IDF v4.0.1 and Rust 1.44.0

jeikabu committed:
ESP-IDF v4.2 Rust ~1.48 LLVM 10.0.1

- Normalize dockerfile to be more similar to https://github.com/reitermarkus/cross-xtensa
- Clean temp files, reduces docker image from 12GB to 8GB
- Remove legacy scripts/templates/etc.
